test/livebook/runtime/node_pool_test.exs

```diff
@@ -0,0 +1,69 @@
+defmodule Livebook.Runtime.NodePoolTest do
+  use ExUnit.Case, async: true
+
+  alias Livebook.Runtime.NodePool
+
+  # Tests for Livebook.Runtime.NodePool
+  #
+  # Note:
+  # 1. We do not spawn actual nodes as it can be time
+  #    intensive (on low spec machines)
+  #
+  # 2. We wait a certain period of time after we mock
```

Comment:
Sorry, I missed the first time around but now we no longer need this wait period. It is guaranteed that every message you send after you send the nodeup message will be processed after the nodeup message, so this comment and all Process.sleep can be removed. :)

Reply:
Right, but we need to wait the buffer period so that the node is added to pool.. And even if I set the buffer time to one microsecond, we should ideally sleep for at least 1 or 2 microseconds so that we can be absolutely sure that the name has been appended.

Reply:
Unfortunately it doesn’t work like that. If both of them sleep, even if one is sleeping for longer, there is no guarantee one will process its messages before the other. The important is rather to guarantee which message arrives to the inbox first, so if you set buffer_time to 0, then it is guaranteed that the message will be placed in the inbox immediately and that the next external request will be processed once the node is added to the pool.

Reply:
Ah... I see what you mean now...

Reply:
I will update the tests..